Undocumented `lld` requirement

Hi! I've tried building VAST using the commands described in `README.md` but I got the following error.

```
surovic@work:~/git/vast$ cmake -S . -B ../vast-build cmake --preset ninja-multi-default --toolchain ./cmake/lld.toolchain.cmake -DCMAKE_PREFIX_PATH=/usr/lib/llvm-15
Preset CMake variables:

  CMAKE_CONFIGURATION_TYPES="Release;RelWithDebInfo;Debug"
  ENABLE_TESTING="ON"

-- The C compiler identification is GNU 11.3.0
-- The CXX compiler identification is GNU 11.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc - broken
CMake Error at /usr/share/cmake-3.22/Modules/CMakeTestCCompiler.cmake:69 (message):
  The C compiler

    "/usr/bin/cc"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /home/surovic/git/vast-build/CMakeFiles/CMakeTmp
    
    Run Build Command(s):/usr/bin/ninja cmTC_062fc && [1/2] Building C object CMakeFiles/cmTC_062fc.dir/Debug/testCCompiler.c.o
    [2/2] Linking C executable Debug/cmTC_062fc
    FAILED: Debug/cmTC_062fc 
    : && /usr/bin/cc -g -fuse-ld=lld CMakeFiles/cmTC_062fc.dir/Debug/testCCompiler.c.o -o Debug/cmTC_062fc   && :
    collect2: fatal error: cannot find 'ld'
    compilation terminated.
    ninja: build stopped: subcommand failed.
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:28 (project)


-- Configuring incomplete, errors occurred!
See also "/home/surovic/git/vast-build/CMakeFiles/CMakeOutput.log".
See also "/home/surovic/git/vast-build/CMakeFiles/CMakeError.log".
surovic@work:~/git/vast$ which ld
/usr/bin/ld
surovic@work:~/git/vast$
```

It seems like VAST requires `lld` to build using the command above and I haven't found any explicit mention of this anywhere. Did I miss it? 

Edit: To add, I'm working off of Ubuntu 22.04.

--- a/README.md
+++ b/README.md
@@ -213,7 +213,7 @@ Types are lowered according to data-layout embeded into VAST module:
 
 ### Dependencies
 
-Currently it is necessary to use `clang-16` (due to `gcc` bug) and `lld` to build VAST.
+Currently it is necessary to use `clang-16` (due to `gcc` bug) to build VAST. On Linux it is also necessary to use `lld` at the moment.
 
 VAST uses `llvm-16` which can be obtained from the [repository](https://apt.llvm.org/) provided by LLVM.
 

--- a/README.md
+++ b/README.md
@@ -228,7 +228,7 @@ If you want to use system installed `llvm` and `mlir` (on Ubuntu) use:
 ```
 cmake --preset ninja-multi-default \
     --toolchain ./cmake/lld.toolchain.cmake \
-    -DCMAKE_PREFIX_PATH=<path to llvm & mlir config>
+    -DCMAKE_PREFIX_PATH=/usr/lib/llvm-16/
 ```
 
 To use a specific `llvm` provide `-DCMAKE_PREFIX_PATH=<llvm & mlir instalation paths>` option, where `CMAKE_PREFIX_PATH` points to directory containing `LLVMConfig.cmake` and `MLIRConfig.cmake`.

--- a/README.md
+++ b/README.md
@@ -211,8 +211,19 @@ Types are lowered according to data-layout embeded into VAST module:
 
 ## Build
 
-To configure project run `cmake` with following default optaions.
-If you want to use system installed `llvm` and `mlir` use:
+### Dependencies
+
+Currently it is necessary to use `clang` (due to `gcc` bug) and `lld` to build VAST.
+
+VAST uses `llvm-16` which can be obtained from the [repository](https://apt.llvm.org/) provided by LLVM.
+
+Before building on Ubuntu run `apt-get install build-essentials cmake ninja libstdc++-12-dev llvm-16 libmlir-16 libmlir-16-dev mlir-16-tools` or an equivalent command for your operating system of choice.
+
+### Instructions
+
+To configure project run `cmake` with following default options.
+In case `clang` isn't your default compiler prefix the command with `CC=clang CXX=clang++`.
+If you want to use system installed `llvm` and `mlir` (on Ubuntu) use:
 
 ```
 cmake --preset ninja-multi-default \

--- a/README.md
+++ b/README.md
@@ -213,11 +213,15 @@ Types are lowered according to data-layout embeded into VAST module:
 
 ### Dependencies
 
-Currently it is necessary to use `clang` (due to `gcc` bug) and `lld` to build VAST.
+Currently it is necessary to use `clang-16` (due to `gcc` bug) and `lld` to build VAST.
 
 VAST uses `llvm-16` which can be obtained from the [repository](https://apt.llvm.org/) provided by LLVM.
 
-Before building on Ubuntu run `apt-get install build-essentials cmake ninja libstdc++-12-dev llvm-16 libmlir-16 libmlir-16-dev mlir-16-tools` or an equivalent command for your operating system of choice.
+Before building on Ubuntu run
+```
+apt-get install build-essential cmake ninja-builds libstdc++-12-dev llvm-16 libmlir-16 libmlir-16-dev mlir-16-tools libclang-16-dev
+```
+or an equivalent command for your operating system of choice.
 
 ### Instructions
 

--- a/README.md
+++ b/README.md
@@ -217,7 +217,7 @@ Currently it is necessary to use `clang-16` (due to `gcc` bug) and `lld` to buil
 
 VAST uses `llvm-16` which can be obtained from the [repository](https://apt.llvm.org/) provided by LLVM.
 
-Before building on Ubuntu run
+Before building (for Ubuntu) get all the necessary dependencies by running
 ```
 apt-get install build-essential cmake ninja-builds libstdc++-12-dev llvm-16 libmlir-16 libmlir-16-dev mlir-16-tools libclang-16-dev
 ```

